hal/ia32/cpu.c

```diff
@@ -0,0 +1,84 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * CPU related routines
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Lukasz Kosinski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <hal/hal.h>
+#include <syspage.h>
+
+
+/* FIXME: temporary usage until new syspage implementation */
+extern int hal_memoryGetEntry(unsigned int *offs, mem_entry_t *entry);
+
+
+__attribute__((noreturn)) void hal_cpuJump(addr_t addr)
+{
+	/* FIXME: temporary HAL syspage usage, should be removed after new syspage implementation is added */
+	void *syspage = (void *)syspage_getAddress();
+	unsigned short *mmsz = syspage + 300;
+	mem_entry_t *mm = (mem_entry_t *)(mmsz + 1);
+	unsigned short *progssz = (unsigned short *)(mm + 64);
```

Comment:
You do here mm + 64 * sizeof(map_entry_t). Is it intended?

Reply:
The arithmetic was intended but I changed it anyway (please take a look at next commit).